### ISSUE TITLE
Contours at lower zoom levels

### ISSUE THREAD

**Issue by fallaciousreasoning** [linz/basemaps#3120]
### User Story

In order to see a more detailed view of the terrain as a user I want to be able to see contours in more detail
instead of the current system, where 20m contours are only visible from zoom level 14, and 100m contours are only visible from zoom level 11.

#### Acceptance Criteria

The current behaviour could stay the same, but it would be great if the tiles included contours slightly more zoomed out - currently the data isn't as useful as the Topo50 series because its hard to get an idea of the steepness of various bits of terrain.

| Basemaps | Topo Series |
|--------|--------|
| ![image](https://github.com/linz/basemaps/assets/7678024/a40d73fb-1f36-49df-bc2e-26744ad34315) | ![image](https://github.com/linz/basemaps/assets/7678024/cf1ae54e-b7a2-479d-9944-da2823fede1d) |
| ![image](https://github.com/linz/basemaps/assets/7678024/541acd07-0b7f-4d29-aaee-ce34b2f0819f) | ![image](https://github.com/linz/basemaps/assets/7678024/15281788-4978-4a26-9ebd-2c7c87057174) |
| ![image](https://github.com/linz/basemaps/assets/7678024/85f9e593-267d-454b-a93e-71137f15ca6b) | ![image](https://github.com/linz/basemaps/assets/7678024/f459b108-d6e8-4390-91fd-54597ee6a299) | 

In each case, the Topo50/Topo250 map shows significantly more elevation detail. If the tiles included this data (but didn't render it) we could customize how detailed the contours should be at each zoom level. Potentially, this could be controlled via some query parameters?

**Comment by dwsilk** [comment]
Hi @fallaciousreasoning, thanks heaps for raising this issue. We are currently working on providing a terrain-rgb service which will allow 3D mapping / dynamic hillshades and theoretically dynamic contour generation (although that may be very resource intensive). This will utilise our increasing coverage of LiDAR data. We'll incorporate your feedback when we look at how to improve our vector tile service with these improved elevation data sources and services.

**Comment by fallaciousreasoning** [comment]
Thanks for all your work on this @dwsilk - the data LINZ provides is super useful :smile:

**Comment by dwsilk** [comment]
Worth noting that once we release the terrain-rgb service, https://github.com/onthegomap/maplibre-contour could be used to incorporate user-configured contours within a MapLibre web map.

See also https://github.com/maplibre/maplibre-style-spec/issues/583 regarding bringing this functionality into the maplibre style spec.

**Comment by fallaciousreasoning** [comment]
Wow, that proposal sounds like exactly what I need - thanks :)

**Comment by dwsilk** [comment]
The terrain-rgb service has now been released: https://basemaps.linz.govt.nz/@-41.3649877,174.6876136,z7.49?i=elevation
Our topographic vector tiles also now include a far more detailed hillshade that is rendered from the terrain-rgb service: https://basemaps.linz.govt.nz/@-41.1332005,175.0810181,z16.96?style=topographic&i=topographic

And maplibre-contour is being used by some of our customers to generate contours from the terrain-rgb service as well.

**Comment by fallaciousreasoning** [comment]
Wow, awesome @dwsilk. Is that data going to be available to download as well? I've been working on generating my own terrain-rgb tiles from the DEM model LINZ provides but I'm definitely not doing an amazing job (I get a lot of artifacts):
https://alpha.topos.nz/#lat=-43.226122&lon=170.948456&zoom=12.7&rotation=0&pitch=0&basemap=topoVector&overlays=hillshade%2Ccontour-source&lla=-43.226571&llo=170.947646&lab=Lauper+Pk+%282428m%29

Any chance you have an mbtiles file available for the terrain-rgb tiles?

**Comment by dwsilk** [comment]
We do not, the terrain-rgb tiles are rendered on-the-fly from Cloud Optimised GeoTIFFs. The source LiDAR survey data is however all available in a public bucket - s3://nz-elevation.

Anyway, it looks like you're rendering contours from our terrain-rgb service now when I click on that link? Looks really fast too!

**Comment by fallaciousreasoning** [comment]
Yeah, using your terrain-rbg service now (it works great). I generated my own tiles first but got way more artifacts than what you're putting out.

**Comment by JBLINZ** [comment]
Hi @fallaciousreasoning Just letting you know that our topographic-v2 styleJson now introduces index contours at z10 and all contours at z13. 

https://basemaps.linz.govt.nz/@-43.4649390,171.1880229,z13.06?style=topographic-v2&i=topographic-v2

We also have [gridded](https://basemaps.linz.govt.nz/@-43.4649390,171.1880229,z13.06?style=topo-raster-gridded&i=topo-raster-gridded) and [gridless](https://basemaps.linz.govt.nz/@-43.4649390,171.1880229,z13.06?style=topo-raster&i=topo-raster) raster topo maps avalible.
